app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,631 @@
+
+Finally, when we have obtained a method handle, we can invoke the underlying method.
+
+<a id="invocation">&nbsp;</a>
+## Method handle invocation
+The invocation can also be done in multiple ways.
+
+All the methods that facilitate invocation eventually funnel down to a single method that is called in the end:
+[`MethodHandle.invokeExact`](javadoc:MethodHandle.invokeExact(Object...)). As the method name suggests, the arguments
+provided to `invokeExact` method must strictly match the method handle's type.
+
+For example, if we invoke a `String.replace` method, the arguments must strictly
+correspond to a `String` return type and two `char` arguments:
+
+```java
+MethodType replaceMethodType = MethodType.methodType(String.class, char.class, char.class);
+MethodHandle replaceMethodHandle = lookup.findVirtual(String.class, "replace", replaceMethodType);
+String result = (String) replaceMethodHandle.invokeExact("dummy", 'd', 'm');
+```
+
+[`MethodHandle.invoke`](javadoc:MethodHandle.invoke(Object...)) is more permissive. It attempts to obtain a new method
+handle with adjusted types that would strictly match the types of provided arguments. After that, it will be able to
+invoke the adjusted method handle using `invokeExact`.
+
+```java
+MethodType replaceMethodType = MethodType.methodType(String.class, char.class, char.class);
+MethodHandle replaceMethodHandle = lookup.findVirtual(String.class, "replace", replaceMethodType);
+String result = (String) replaceMethodHandle.invoke((Object)"dummy", (Object)'d', (Object)'m'); // would fail with `invokeExact`
+```
+
+One other alternative to invoke a method handle is to use [`MethodHandle.invokeWithArguments`](javadoc:invokeWithArguments(Object...)).
+The result of this method invocation is equivalent to `invoke`, with the only difference that all the arguments can be
+provided as an array or list of objects.
+
+One interesting feature of this method is that if the number of provided arguments exceeds the expected number, all the
+leftover arguments will be squashed into the last argument, which will be treated as an array.
+
+<a id="fields">&nbsp;</a>
+## Accessing fields
+It is possible to create method handles that have read or write access to fields. For instance fields, this is
+facilitated by [`findGetter`](javadoc:MethodHandles.Lookup.findGetter(Class,String,Class)) and
+[`findSetter`](javadoc:MethodHandles.Lookup.findSetter(Class,String,Class)) methods, and for static fields, by
+[`findStaticGetter`](javadoc:MethodHandles.Lookup.findStaticGetter(Class,String,Class)) and
+[`findStaticSetter`](javadoc:MethodHandles.Lookup.findStaticSetter(Class,String,Class)) methods. We don't need to provide
+a `MethodType` instance; instead, we should provide a single type, which is the type of the field.
+
+For example, if we have a static field `magic` in our `Example` class:
+
+```java
+private static String magic = "initial value static field";
+```
+
+Given that we have created a `Lookup` object
```

Comment:
Just a note: In contrast to other code examples, that line/sentence ends without any punctuation.